templates/_helpers.tpl

```diff
@@ -274,7 +299,12 @@ Sets the injector node selector for pod placement
 {{- define "injector.nodeselector" -}}
   {{- if .Values.injector.nodeSelector }}
       nodeSelector:
-        {{ tpl .Values.injector.nodeSelector . | indent 8 | trim }}
+        {{- $tp := typeOf .Values.injector.nodeSelector }}
+        {{- if eq $tp "string" }}
+          {{- tpl .Values.injector.nodeSelector . | indent 8 }}
```

Comment:
I'm curious why `trim` was removed from all the string cases.

Reply:
I was getting YAML parse errors in the tests with `trim`, but I see now that we were getting extra white space trimmed by the Go template itself. 

The following change allows us to use `trim` without breaking the YAML.
```
@@ -551,7 +551,7 @@ Sets the injector toleration for pod placement
       tolerations:
       {{- $tp := typeOf .Values.csi.pod.tolerations }}
       {{- if eq $tp "string" }}
-        {{- tpl .Values.csi.pod.tolerations . | nindent 8 }}
+        {{ tpl .Values.csi.pod.tolerations . | nindent 8 | trim }}
       {{- else }}
         {{- toYaml .Values.csi.pod.tolerations | nindent 8 }}
       {{- end }}

```